src/components/ImageEditor.vue

```diff
@@ -179,7 +181,20 @@ export default {
 					emit('files:node:created', { fileid: parseInt(response?.headers?.['oc-fileid']?.split('oc')[0]) || null })
 				} else {
 					this.$emit('updated')
-					emit('files:node:updated', { fileid: this.fileid })
+					const updatedFile = await rawStat(origin, decodeURI(pathname))
+
+					const node = new Node({
+						id: Number.parseInt(this.fileid),
+						source: this.src,
+						mtime: new Date(updatedFile.lastmod),
+						...updatedFile,
+						attributes: {
+							...updatedFile,
+							...updatedFile.props,
+						},
+					})
+
+					emit('files:node:updated', { fileid: this.fileid, node })
```

Comment:
Node directly?

```suggestion
					emit('files:node:updated', node)
```

Reply:
https://nextcloud.github.io/nextcloud-files/classes/File.html#fileid

Reply:
I didn't want to break compatibility with existing listeners.
Also, extending the object allows to easily add more properties in the future if needed.

Reply:
Is there any existing ones for that event? I think I introduced it for nc27 only? :thinking:

Reply:
Apparently from you :sweat_smile: : https://github.com/nextcloud/server/blob/d9e9d4e123f5b74378b9ba2fa2082eca205f3526/apps/files/src/components/NavigationQuota.vue#L86
And the Photos one: https://github.com/nextcloud/viewer/blob/64ce3fa4b542da00cb16f4878054873b54e7d6aa/src/views/Viewer.vue#L470

Reply:
> Apparently from you  : [nextcloud/server@`d9e9d4e`/apps/files/src/components/NavigationQuota.vue#L86](https://github.com/nextcloud/server/blob/d9e9d4e123f5b74378b9ba2fa2082eca205f3526/apps/files/src/components/NavigationQuota.vue#L86)

which doesn't use the payload, so we're good :)

Reply:
You don't think it is worth having the possibility to easily extend the event payoad in the futur?

Reply:
I think the payload is the node itself. Any more data should be in the Node attributes :)

Reply:
Reformulating:

You think the payload should be the node itself. And that we won't need data that do not belong in the Node attributes.

I have no use case in mind, but it is not hard to imagine that one day we might need more context than the node info. Like its previous state, the time of the request, or where it was made. 

I am starting to be against having the node as the payload, as I feel like we are restricting ourselves for not reason.

Reply:
cc @juliushaertl @ChristophWurst @susnux @max-nextcloud, any opinions?

```ts
emit('files:node:updated', { fileid: string, node: Node })
OR
emit('files:node:updated', node: Node)
```

Reply:
Not sure if there are already other events, because I think keeping it consistent is more important, so either directly the updated node vs context objects `{ node: Node }`.

If not then I would go for just the node, and either way as this is new and not used currently I would **not** keep the `fileid` as it is duplicated, you can simply query it from the node (`node.fileid`).

Reply:
The question was only about `node` vs `{node}` :)
I personally think the payload should be the subject of the event, since we're talking about a node, then we shouldn't have to search for the node within an object.

Please also note that we already have some events on 27 using node directly.
https://github.com/nextcloud/server/blob/a3fe1adb03ad5814298e4f18d14d98f85096fe7c/apps/files/src/actions/deleteAction.ts#L54

So that means we would be  breaking things for 28

Reply:
Let's get this PR moving, I switched to `node` as the payload of the event.